apps/src/code-studio/components/ExportDialog.jsx

```diff
@@ -0,0 +1,31 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {connect} from 'react-redux';
+import {SignInState} from '@cdo/apps/code-studio/progressRedux';
+import ExportAllowedDialog from './ExportAllowedDialog';
+import ShareDisallowedDialog from './ShareDisallowedDialog';
+
+class ExportDialog extends Component {
+  static propTypes = {
+    signInState: PropTypes.oneOf(Object.values(SignInState)),
+    isProjectLevel: PropTypes.bool.isRequired
+  };
+
+  render() {
+    const {signInState, isProjectLevel, ...otherProps} = this.props;
+    // If we're on a project level (i.e. /projects/appname), always show signed
+    // in version of the dialog
+
+    if (signInState === SignInState.SignedIn || isProjectLevel) {
+      return <ExportAllowedDialog {...otherProps} />;
+    }
+
+    return <ShareDisallowedDialog />; // TODO: re-use or create ExportDisallowedDialog?
```

Comment:
I'd very much prefer to either create an `ExportDisallowedDialog` or rename `ShareDisallowedDialog` to something more generic

Reply:
Ok, I figured out a way to eliminate the extra component altogether. Trying to mimic the layers of warnings in the share dialog has been a little challenging. There are so many that have been bolted on!